store/src/java/com/zimbra/cs/redolog/RedoLogManager.java

```diff
@@ -380,7 +402,8 @@ public void log(RedoableOp op, boolean synchronous) {
 
         logOnly(op, synchronous);
 
-        if (isRolloverNeeded(false))
+        //rollover is needed only when use FileLogWriter mechanism
+        if (isRolloverNeeded(false) && (getLogWriter() instanceof FileLogWriter))
```

Comment:
Change to `!(mLogWriter instanceof DbLogWriter)`. Use negation to ensure that we are changing the behavior for only the new `DbLogWriter`.